test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -6793,6 +6793,218 @@ BOOST_AUTO_TEST_CASE(cleanup_bytes_types)
 	BOOST_CHECK(callContractFunction("f(bytes2,uint16)", string("abc"), u256(0x040102)) == encodeArgs(0));
 }
 
+BOOST_AUTO_TEST_CASE(const_fixed_type_return)
+{
+	char const* sourceCode = R"(
+		contract C {
+			fixed constant a = 3.5;
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_function_return)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() constant returns (fixed a) {
+				return 325.5;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_function_multi_return)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (fixed a, fixed b, fixed c) {
+				a = 325.5;
+				b = 450.25;
+				c = 23924029.9375;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(negative_exponent)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (fixed a) {
+				a = 2 ** -5;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK(callContractFunction("a()") == encodeArgs(0));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_arguments)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed, fixed) {
+				return (a, b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_to_int)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a) returns (uint b) {
+				b = a;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_addition)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a + b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_subtraction)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a - b);
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	BOOST_CHECK();
+}
+
+BOOST_AUTO_TEST_CASE(fixed_type_multiplication)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f(fixed a, fixed b) returns (fixed) {
+				return (a * b);
```

Comment:
Multiplication is very interesting with types that do not need all the 256 bits, please also add tests for that!